kore/src/Kore/Builtin/InternalBytes.hs

```diff
@@ -403,6 +411,44 @@ int2bytes len int end =
     word8 :: Integer -> Word8
     word8 = toEnum . fromEnum
 
+evalBytes2int :: Builtin.Function
+evalBytes2int =
+    Builtin.functionEvaluator evalBytes2int0
+  where
+    evalBytes2int0 :: Builtin.FunctionImplementation
+    evalBytes2int0 resultSort [bytes, end, sign] =
+        Builtin.getAttemptedAxiom $ do
+            end' <-
+                case end of
+                    Endianness_ endianness -> return endianness
+                    _                      -> empty
+            sign' <-
+                case sign of
+                    Signedness_ signedness -> return signedness
+                    _                      -> empty
+            bytes' <- expectBuiltinBytes bytes2intKey bytes
+            let result = bytes2int bytes' end' sign'
+            Builtin.appliedFunction $ Int.asPattern resultSort result
+    evalBytes2int0 _ _ = Builtin.wrongArity bytes2intKey
+
+bytes2int :: ByteString -> Endianness -> Signedness -> Integer
+bytes2int bytes end sign =
+    case sign of
+        Unsigned _ -> unsigned
+        Signed   _
+          | 2 * unsigned > modulus -> unsigned - modulus
+          | otherwise              -> unsigned
+  where
+    (modulus, unsigned) = ByteString.foldl' go (1, 0) littleEndian
+    go (!place, !acc) byte =
```

Comment:
Why this strictness annotation? I'd assume that everything is evaluated when putting the result in the term anyway.

I mean: I don't think there's anything wrong with this, I'm just curious.